core/src/subgraph/runner.rs

```diff
@@ -899,12 +900,14 @@ where
                 .observe(block.trigger_count() as f64);
         }
 
+        let ptr_updates_section = self.metrics.stream.stopwatch.start_section("ptr_updates");
+
         if block.trigger_count() == 0
```

Comment:
An idea, lets wrap this whole if block in an outer if with the condition:
`if self.state.skip_ptr_updates_timer.elapsed() && !self.state.synced`
This way, if the timer hasn't elapsed or the subgraph is synced, we won't even bother with resetting the timer and updating the cached head.

Reply:
Just pushed a commit, hope the change made is what you suggested

Reply:
hmm, intereseting i thought when the variable that is assigned to the section goes out of scope the timer is reset.

Reply:
I think that the check for `!self.state.synced` should be separated and outer one, and then the else branch of the check for `self.state.skip_ptr_updates_timer.elapsed()` should reset the timer too.